compiler/src/main/scala/org/scalajs/core/compiler/GenJSCode.scala

```diff
@@ -2802,14 +2802,34 @@ abstract class GenJSCode extends plugins.PluginComponent
          *   {name1: arg1, name2: arg2, ... }
          */
 
+        def warnIfDuplicatedKey(pairs: List[(js.StringLiteral, js.Tree)]): Unit = {
+          val allKeys = pairs.collect { case (js.StringLiteral(keyName), _) => keyName }
+          val keyCounts = allKeys.distinct.map(key => key -> allKeys.count(_ == key))
+          val duplicateKeyCounts = keyCounts.filter(1 < _._2)
```

Comment:
`groupBy` is annoying in this case because, as any `Map`, it does not guarantee the order of the keys in the resulting collection. Which means error messages can randomly change.